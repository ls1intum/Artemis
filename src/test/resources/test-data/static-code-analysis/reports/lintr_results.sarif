{
  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
  "version": "2.1.0",
  "runs": [
    {
      "tool": {
        "driver": {
          "name": "lintr",
          "informationUri": "https://lintr.r-lib.org/",
          "version": "3.2.0",
          "rules": [
            {
              "id": "object_name_linter",
              "fullDescription": {
                "text": "Variable and function name style should match snake_case or symbols."
              },
              "defaultConfiguration": {
                "level": "note"
              }
            }
          ]
        }
      },
      "columnKind": "utf16CodeUnits",
      "originalUriBaseIds": {
        "ROOTPATH": {
          "uri": "file:///var/tmp/testing-dir/assignment/"
        }
      },
      "results": [
        {
          "ruleId": "object_name_linter",
          "ruleIndex": 0,
          "message": {
            "text": "Variable and function name style should match snake_case or symbols."
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "R/hello.R",
                  "uriBaseId": "ROOTPATH"
                },
                "region": {
                  "startLine": 16,
                  "startColumn": 1,
                  "snippet": {
                    "text": "helloWorld <- function() {"
                  }
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
