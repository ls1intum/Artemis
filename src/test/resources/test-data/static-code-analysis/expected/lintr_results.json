{
  "tool": "LINTR",
  "issues": [
    {
      "filePath": "R/hello.R",
      "startLine": 16,
      "endLine": 16,
      "startColumn": 1,
      "endColumn": 2,
      "rule": "object_name_linter",
      "category": "Lint",
      "message": "Variable and function name style should match snake_case or symbols.",
      "priority": "note"
    }
  ]
}
